tests/bdd/features/volume_group/create/test_create.py

```diff
@@ -0,0 +1,267 @@
+def three_volumes_created_with_three_replicas():
+    """3 volumes should be created with 3 replica each."""
+    filtered_volumes = get_volume_group_volumes()
+    assert len(filtered_volumes) == 3
+    for volume in filtered_volumes:
+        assert volume.spec.num_replicas == 3
+
+
+@then("each replica of individual volume should be on a different node")
+def check_replicas_different_node_individual_volume():
+    """each replica of individual volume should be on a different node."""
+    filtered_volumes = get_volume_group_volumes()
+    for volume in filtered_volumes:
+        replica_topology = volume.state.replica_topology
+        node_locations = [replica.node for replica in replica_topology.values()]
+        assert len(set(node_locations)) == len(replica_topology)
+
+
+@then("each replica should reside on a different node")
+def check_replicas_different_node():
+    """each replica should reside on a different node."""
+    filtered_volumes = get_volume_group_volumes()
+    already_encountered_nodes = set()
+    for volume in filtered_volumes:
+        replica_topology = volume.state.replica_topology
+        for replica in replica_topology.values():
+            node = replica.node
+            if node in already_encountered_nodes:
+                assert False
+            already_encountered_nodes.add(node)
+
+
+@then("each replica should reside on a different pool")
+def check_replicas_different_pool():
+    """each replica should reside on a different pool."""
+    filtered_volumes = get_volume_group_volumes()
+    already_encountered_pools = set()
+    for volume in filtered_volumes:
+        replica_topology = volume.state.replica_topology
+        for replica in replica_topology.values():
+            pool = replica.pool
+            if pool in already_encountered_pools:
+                assert False
+            already_encountered_pools.add(pool)
+
+
+@then("pools can have more than one replicas all belonging to different volumes")
+def check_pool_replicas_different_volumes():
+    """pools can have more than one replicas all belonging to different volumes."""
+    filtered_volumes = get_volume_group_volumes()
+    pool_replicas = {}
+    for volume in filtered_volumes:
+        replica_topology = volume.state.replica_topology
+        for replica in replica_topology.values():
+            pool = replica.pool
+            if pool in pool_replicas:
+                if volume.spec.uuid in pool_replicas[pool]:
+                    assert False
+                else:
+                    pool_replicas[pool].append(volume.spec.uuid)
+            else:
+                pool_replicas[pool] = [volume.spec.uuid]
+
+
+# HELPER METHODS
+
+# Creates the pools based on given nodes and counts.
+def create_node_pools(node_pool_map: Dict[str, int]):
+    for node_name, num_pools in node_pool_map.items():
+        for i in range(num_pools):
+            pool_name = f"pool_{uuid4().hex}"
```

Comment:
Small nit: might be worth formatting pool names using index so it’s more human friendly if we need to debug

Reply:
Done.